File: web-simulator/js/apps/pomodoro.js
/** TinyOLED Desktop — Pomodoro Timer */
export class PomodoroApp {
  static NAME='pomodoro'; static LABEL='Pomo'; static ICON='timer';
  static WORK=25*60; static BREAK=5*60;
  constructor(onExit,notify) {
    this.onExit=onExit; this.notify=notify||(()=>{});
    this.running=false; this.working=true;
    this.remaining=PomodoroApp.WORK; this.sessions=0; this._last=0;
  }
  onUp(){} onDown(){}
  onSel() {
    if(!this.running){this.running=true;this._last=performance.now()/1000;}
    else this.running=false;
  }
  onLong(){this.onExit();}
  update() {
    if(!this.running) return;
    const now=performance.now()/1000, dt=now-this._last; this._last=now;
    this.remaining-=dt;
    if(this.remaining<=0) {
      if(this.working){this.sessions++;this.notify(`Mola! (${this.sessions})`);this.remaining=PomodoroApp.BREAK;}
      else{this.notify('Calismaya basla!');this.remaining=PomodoroApp.WORK;}
      this.working=!this.working;
    }
  }
  draw(fb) {
    fb.icon('timer',1,10); fb.text('Pomodoro',12,10); fb.hline(0,18,128);
    const mins=Math.floor(this.remaining/60), secs=Math.floor(this.remaining%60);
    fb.textCentered(`${String(mins).padStart(2,'0')}:${String(secs).padStart(2,'0')}`,30);
    const total=this.working?PomodoroApp.WORK:PomodoroApp.BREAK;
    fb.progressBar(10,42,108,6,total-this.remaining,total);
    fb.text(this.working?'CALISMA':'MOLA',2,52);
    fb.text(`#${this.sessions}`,60,52);
    fb.text(this.running?'DUR':'BASLA',90,52);
  }
}

File: apps/pomodoro.py
"""TinyOLED Desktop — Pomodoro Zamanlayıcı"""
import time
from core.framebuffer import Framebuffer

class PomodoroApp:
    NAME="pomodoro"; LABEL="Pomo"; ICON="timer"
    WORK=25*60; BREAK=5*60
    def __init__(self, on_exit, notify=None):
        self.on_exit=on_exit; self.notify=notify or (lambda m:None)
        self.running=False; self.working=True; self.remaining=self.WORK; self.sessions=0; self._last=0
    def on_up(self): pass
    def on_down(self): pass
    def on_sel(self):
        if not self.running: self.running=True; self._last=time.monotonic()
        else: self.running=False
    def on_long(self): self.on_exit()
    def update(self):
        if not self.running: return
        now=time.monotonic(); dt=now-self._last; self._last=now; self.remaining-=dt
        if self.remaining<=0:
            if self.working: self.sessions+=1; self.notify(f"Mola! ({self.sessions})"); self.remaining=self.BREAK
            else: self.notify("Calismaya basla!"); self.remaining=self.WORK
            self.working=not self.working
    def draw(self, fb: Framebuffer):
        fb.icon("timer",1,10); fb.text("Pomodoro",12,10); fb.hline(0,18,128)
        mins=int(self.remaining)//60; secs=int(self.remaining)%60
        fb.text_centered(f"{mins:02d}:{secs:02d}",30)
        total=self.WORK if self.working else self.BREAK
        fb.progress_bar(10,42,108,6,total-self.remaining,max_val=total)
        fb.text("CALISMA" if self.working else "MOLA",2,52)
        fb.text(f"#{self.sessions}",60,52)
        fb.text("DUR" if self.running else "BASLA",90,52)
